ee/desktop/runner/runner_linux.go

```diff
@@ -132,3 +156,63 @@ func (r *DesktopUsersProcessesRunner) userEnvVars(uid string) map[string]string
 
 	return envVars
 }
+
+func (r *DesktopUsersProcessesRunner) displayFromXwayland(uid int32) string {
+	//For wayland, DISPLAY is not included in loginctl show-session output -- in GNOME,
+	// Mutter spawns Xwayland and sets $DISPLAY at the same time. Find $DISPLAY by finding
+	// the Xwayland process and examining its args.
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	processes, err := process.ProcessesWithContext(ctx)
+	if err != nil {
+		level.Debug(r.logger).Log(
+			"msg", "could not query processes to find Xwayland process",
+			"err", err,
+		)
+		return defaultDisplay
+	}
+
+	for _, p := range processes {
+		cmdline, err := p.CmdlineWithContext(ctx)
+		if err != nil {
+			level.Debug(r.logger).Log(
+				"msg", "could not get cmdline slice for process",
+				"err", err,
+			)
+			continue
+		}
+
+		if !strings.Contains(cmdline, "Xwayland") {
+			continue
+		}
+
+		// We have an Xwayland process -- check to make sure it's for our running user
+		uids, err := p.UidsWithContext(ctx)
+		if err != nil {
+			level.Debug(r.logger).Log(
+				"msg", "could not get uids for process",
+				"err", err,
+			)
+			continue
+		}
+
+		for _, procUid := range uids {
+			if procUid != uid {
+				continue
+			}
+
+			// We have a match! Grab the display value. The xwayland process looks like:
+			// /usr/bin/Xwayland :0 -rootless -noreset -accessx -core -auth /run/user/1000/.mutter-Xwaylandauth.ROP401 -listen 4 -listen 5 -displayfd 6 -initfd 7
+			cmdlineArgs := strings.Split(cmdline, " ")
+			if len(cmdlineArgs) < 2 {
+				// Process is somehow malformed or not what we're looking for -- stop evaluating it
+				break
```

Comment:
not continue?

Reply:
I'm mostly thinking about all the ugly corner cases.

What if the user has 2 displays, :0 and :1? Maybe we ignore that.

What if they have other processes running called `Xwayland`? (maybe `grep Xwayland`)? Is it better to continue and see if we find one that matches what we want? 

I think return-first-match makes sense. But then on not-match, continue seems okay.

Reply:
I've updated this so that it makes it clearer what we're doing -- we _are_ doing return-first-match and continue on no-match.

I don't have another process running called `Xwayland`, at least, but yeah I'm a little unsure about the edge cases too.

Reply:
I think we only have imperfect choices. So this seems fine, and we'll iterate as we need